x-pack/plugins/security_solution/public/management/services/exceptions_list/exceptions_list_api_client.test.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreStart, HttpSetup } from 'kibana/public';
+import { CreateExceptionListSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { EXCEPTION_LIST_ITEM_URL, EXCEPTION_LIST_URL } from '@kbn/securitysolution-list-constants';
+import { coreMock } from '../../../../../../../src/core/public/mocks';
+import { ExceptionsListItemGenerator } from '../../../../common/endpoint/data_generators/exceptions_list_item_generator';
+import { ExceptionsListApiClient } from './exceptions_list_api_client';
+
+const getFakeListId: () => string = () => 'FAKE_LIST_ID';
+const getFakeListDefinition: () => CreateExceptionListSchema = () => ({
+  name: 'FAKE_LIST_NAME',
+  namespace_type: 'agnostic',
+  description: 'FAKE_LIST_DESCRIPTION',
+  list_id: getFakeListId(),
+  type: 'endpoint',
+});
+const getQueryParams = () => ({
+  page: 1,
+  perPage: 10,
+  filter: 'this is a KQL filter',
+  sortField: 'id',
+  sortOrder: 'asc',
+});
+
+describe('Exceptions List Api Client', () => {
+  let fakeCoreStart: jest.Mocked<CoreStart>;
+  let fakeHttpServices: jest.Mocked<HttpSetup>;
+  let getInstance: () => ExceptionsListApiClient;
+
+  // Initialize mocks once as the ExceptionsListApiClient is a singleton
+  beforeAll(() => {
+    fakeCoreStart = coreMock.createStart({ basePath: '/mock' });
+    fakeHttpServices = fakeCoreStart.http as jest.Mocked<HttpSetup>;
+  });
+
+  // Clear every function mock on each execution
+  beforeEach(() => {
+    fakeHttpServices.post.mockClear();
+    fakeHttpServices.get.mockClear();
+    fakeHttpServices.put.mockClear();
+    fakeHttpServices.delete.mockClear();
+    getInstance = () =>
+      ExceptionsListApiClient.getInstance(
+        fakeHttpServices,
+        getFakeListId(),
+        getFakeListDefinition()
+      );
+  });
+
```

Comment:
can you add a test for when multiple instance of the class for the same list are instantiated to ensure that the "createExceptionList()` is only called once per list id?  You might need to to mock each HTTP response to have a different delay so that they resolve in a predictable sequential order

Reply:
I think this is already tested in the `Wen getting an instance` describe block. Also it's tested indirectly in `Create method calls http.post with params` when ensuring the post method was called once (the instance is already created).

Reply:
yeah, but have you tested the constructor instantiation 😄  

```typescript
const inst1 = new ExceptionsListApiClient('for trusted apps');

const inst2 = new ExceptionsListApiClient('for trusted apps');

const inst3 = new ExceptionsListApiClient('for trusted apps');
```